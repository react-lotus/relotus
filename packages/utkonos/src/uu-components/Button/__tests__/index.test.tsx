import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter, Link as RouterLink } from 'react-router-dom';

import { Button } from '..';

describe('Button component', () => {
  it('render correctly', () => {
    expect(() => render(<Button>Follow</Button>)).not.toThrow();
  });

  it('should render empty button without errors', () => {
    const { getByRole } = render(
      <Button role="button">
        {null}
        {undefined}
      </Button>,
    );
    expect(getByRole('button')).toMatchSnapshot();
  });

  it('should render small button', () => {
    const { getByText } = render(<Button small>Small</Button>);
    const button = getByText('Small');

    expect(button).toHaveClass('uu-button_small');
    expect(button).toMatchSnapshot();
  });

  it('should not clickable when button is loading', () => {
    const onClick = jest.fn();
    const text = 'loading';
    const { getByText } = render(
      <Button loading onClick={onClick}>
        {text}
      </Button>,
    );
    const button = getByText(text);

    fireEvent.click(button);

    expect(onClick).not.toHaveBeenCalled();
  });

  it('should button have a loading animation', () => {
    const text = 'loading';
    const { getByText } = render(<Button loading>{text}</Button>);
    const button = getByText(text);

    expect(button).toHaveClass('uu-button_loading');
    expect(button).toMatchSnapshot();
  });

  it('should not clickable when button is disabled', () => {
    const onClick = jest.fn();
    const text = 'disabled';
    const { getByText } = render(
      <Button disabled onClick={onClick}>
        {text}
      </Button>,
    );
    const button = getByText(text);

    fireEvent.click(button);

    expect(onClick).not.toHaveBeenCalled();
  });

  it('should render as tag <span />', () => {
    const text = 'span';
    const { getByText } = render(<Button as="span">{text}</Button>);

    expect(getByText(text).tagName).toBe('SPAN');
  });

  it('should render as tag <a />', () => {
    const text = 'a link';
    const { getByText } = render(<Button as="a">{text}</Button>);

    expect(getByText(text).tagName).toBe('A');
  });

  it('should render as RouterLink', () => {
    const text = 'Router Link';
    const { getByText } = render(
      <BrowserRouter>
        <Button as={RouterLink} to="/">
          {text}
        </Button>
      </BrowserRouter>,
    );
    const linkedButton = getByText(text);

    expect(linkedButton.tagName).toBe('A');
    expect(linkedButton).toMatchSnapshot();
  });
});
