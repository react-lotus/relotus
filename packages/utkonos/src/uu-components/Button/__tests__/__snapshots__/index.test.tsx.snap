// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should button have a loading animation 1`] = `
<button
  class="uu-button uu-button_view_primary uu-button_loading"
  disabled=""
  type="button"
>
  loading
</button>
`;

exports[`Button component should render as RouterLink 1`] = `
<a
  class="uu-button uu-button_view_primary"
  href="/"
>
  Router Link
</a>
`;

exports[`Button component should render empty button without errors 1`] = `
<button
  class="uu-button uu-button_view_primary"
  role="button"
  type="button"
/>
`;

exports[`Button component should render small button 1`] = `
<button
  class="uu-button uu-button_view_primary uu-button_small"
  type="button"
>
  Small
</button>
`;
